src/main/java/it/pagopa/pn/bff/service/senderdashboard/SenderDashboardService.java

```diff
@@ -0,0 +1,87 @@
+package it.pagopa.pn.bff.service.senderdashboard;
+
+import it.pagopa.pn.bff.exceptions.PnBffException;
+import it.pagopa.pn.bff.exceptions.PnBffExceptionCodes;
+import it.pagopa.pn.bff.generated.openapi.server.v1.dto.BffSenderDashboardDataResponse;
+import it.pagopa.pn.bff.service.senderdashboard.resources.DatalakeS3Resource;
+import it.pagopa.pn.bff.service.senderdashboard.exceptions.SenderNotFoundException;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Mono;
+
+import java.time.LocalDate;
+import java.util.Objects;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class SenderDashboardService {
+
+    private final DatalakeS3Resource datalakeResource;
+
+    /**
+     * Fetches dashboard data for the specified sender.
+     *
+     * @param cxType    Public Administration Type
+     * @param cxId      Public Administration id
+     * @param pathCxType Public Administration Type
+     * @param pathCxId  Public Administration id
+     * @param startDate the start date for the data request
+     * @param endDate   the end date for the data request.
+     * @return a {@link Mono} emitting the {@link BffSenderDashboardDataResponse}.
+     */
+    public Mono<BffSenderDashboardDataResponse> getDashboardData(
+            String cxType,
+            String cxId,
+            String pathCxType,
+            String pathCxId,
+            LocalDate startDate,
+            LocalDate endDate) {
+
+        // Path cxType and cxId mismatch
+        if(!Objects.equals(cxType, pathCxType) || !Objects.equals(cxId, pathCxId)) {
+            return Mono.error(
+                    new PnBffException(
+                        "cxType and cxId mismatch",
+                        "The provided cxType and cxId do not match the values in the headers " +
+                                "x-pagopa-pn-cx-type and x-pagopa-pn-cx-id",
+                        HttpStatus.BAD_REQUEST.value(),
+                        PnBffExceptionCodes.ERROR_CODE_PN_GENERIC_ERROR
```

Comment:
I'm not sure about the use of PnBffExceptionCodes.ERROR_CODE_PN_GENERIC_ERROR here and for the others errors below. The use of this error code is discouraged. Check the other micro-services to see how they manage similar cases